using-mobile-extensions/adobe-target/target-api-reference.md

```diff
@@ -643,34 +1145,119 @@ The following method can be used to create an instance of a Target prefetch obje
+
+```
+
+
+
 ```text
+@deprecated
 + (nonnull instancetype) prefetchObjectWithName: (nonnull NSString*) name                                 mboxParameters: (nullable NSDictionary*) mboxParameters;
 ```
 
-The following method can be used to create an instance of a Target prefetch object that might be used to make a batch request to the configured Target server to prefetch content for mbox locations:
+### ACPTargetParameters
+
+This class contains the parameter dictionary for mbox parameters, profile parameters, and ACPTargetOrder object as well as ACPTargetProduct object:
+
+```objectivec
+@interface ACPTargetParameters : NSObject
+
+///Dictionary containing key-value pairs of parameters
+@property(nonatomic, strong, nullable) NSDictionary<NSString*, NSString*>* parameters;
+
+///Dictionary containing key-value pairs of profile parameters
+@property(nonatomic, strong, nullable) NSDictionary<NSString*, NSString*>* profileParameters;
+
+///ACPTargetOrder object
+@property(nonatomic, strong, nullable) ACPTargetOrder* order;
+
+///ACPTargetProduct object
+@property(nonatomic, strong, nullable) ACPTargetProduct* product;
+@end
+```
+
+The following method can be used to create an instance of a Target parameters object that can be used when invoking multiple APIs:
+
+```
++ (nonnull instancetype) targetParametersWithParameters: (nullable NSDictionary*) parameters
+                                      profileParameters: (nullable NSDictionary*) profileParameters
+                                                product: (nullable ACPTargetProduct*) product
+                                                  order: (nullable ACPTargetOrder*) order;
+
+```
+### ACPTargetOrder
+
+This class contains orderId, total and array for purchasedProductIds present in order parameters:
+
+```objectivec
+@interface ACPTargetOrder : NSObject
+
+///Order ID
+@property(nonatomic, strong, nonnull) NSString* orderId;
+
+///Order total
+@property(nonatomic, strong, nullable) NSNumber* total;
+
+///Array of Purchased Product Ids
+@property(nonatomic, strong, nullable) NSArray<NSString*>* purchasedProductIds;
+@end
+```
+
+The following method can be used to create an instance of a Target order object that can be used when invoking multiple APIs:
+
+```
++ (nonnull instancetype) targetOrderWithId: (nonnull NSString*) orderId
+                                     total: (nullable NSNumber*) total
+                       purchasedProductIds: (nullable NSArray <NSString*>*) purchasedProductIds;
 
-```text
-+ (nonnull instancetype) prefetchObjectWithName: (nonnull NSString*) name
-                                 mboxParameters: (nullable NSDictionary*) mboxParamete
 ```
+### ACPTargetProduct
+
+This class contains productId and categoryId present in product parameters:
+
+```objectivec
+@interface ACPTargetProduct : NSObject
+
+///Product ID
+@property(nonatomic, strong, nullable) NSString* productId;
```

Comment:
Shouldn't productId be `nonnull`, just like orderId?

Reply:
Ideally, it should have been. But i copied this code from iOS source files, and it is `nullable`
Should i change the documentation or leave as it is?